crates/wit-bindgen/src/lib.rs

```diff
@@ -1535,28 +1535,38 @@ impl Wasmtime {
                 )
             } else {
                 uwriteln!(
-                src,
-                "{inst}.resource_async(
-                    \"{name}\",
-                    {wt}::component::ResourceType::host::<{camel}>(),
-                    move |mut store, rep| {{
-                        {wt}::component::__internal::Box::new(async move {{
-                            Host{camel}::drop(&mut host_getter(store.data_mut()), {wt}::component::Resource::new_own(rep)).await
-                        }})
-                    }},
-                )?;"
-            )
+                    src,
+                    "{inst}.resource_async(
+                        \"{name}\",
+                        {wt}::component::ResourceType::host::<{camel}>(),
+                        move |mut store, rep| {{
+                            {wt}::component::__internal::Box::new(async move {{
+                                Host{camel}::drop(&mut host_getter(store.data_mut()), {wt}::component::Resource::new_own(rep)).await
+                            }})
+                        }},
+                    )?;"
+                )
             }
         } else {
+            let (first_arg, trait_suffix) = if flags.contains(FunctionFlags::STORE) {
+                (
+                    format!("{wt}::component::Access::new(store, host_getter)"),
```

Comment:
`Access` constructed this way does not have an `Instance`, which causes the panic in #11651